plugins/inputs/prometheus/prometheus.go

```diff
@@ -124,6 +137,23 @@ func (p *Prometheus) Init() error {
 	if p.MetricVersion != 2 {
 		p.Log.Warnf("Use of deprecated configuration: 'metric_version = 1'; please update to 'metric_version = 2'")
 	}
+
+	if len(p.KubernetesLabelSelector) > 0 {
+		_, err := labels.Parse(p.KubernetesLabelSelector)
+		if err != nil {
+			p.Log.Errorf("label selector validation failed %q: %v", p.KubernetesLabelSelector, err)
+			return err
```

Comment:
To avoid double logging, Telegraf logs any errors returned from this function, only return the error here without logging.  Include the extra context with the error:
```
return fmt.Errorf("label selector validation failed %q: %v", p.KubernetesLabelSelector, err)
```